000-vip-init.php

```diff
@@ -1,8 +1,46 @@
 <?php
 
+
+/**
+ * Log and print a message, then die with an HTTP 500 response
+ *
+ * @param string $msg The message to print
+ */
+function wpcomvip_healthcheck_fail( $msg ) {
+        // Our cache interprets a 500 response as unhealthy
+        status_header( 500 );
+        error_log( $msg );
+        die( $msg );
+}
+
+/**
+ * Exercise the database and object cache
+ * to ensure both are working.
+ */
+function wpcomvip_run_healthcheck() {
+        // Exercise the database by checking for a non-autoloaded option
+        global $wpdb;
+        $sql = "SELECT option_value FROM $wpdb->options WHERE option_name = 'wpcomvip_db_healthcheck'";
+        $db_check = $wpdb->get_var( $sql );
+        // The write only happens when the DB doesn't have the entry, which
+        // should just be the first time this check runs on a given database
+        if ( is_null( $db_check ) ) {
+                add_option( 'wpcomvip_db_healthcheck', true, null, 'no' );
+                $db_check = $wpdb->get_var( $sql );
+        }
```

Comment:
If we just care about verifying that the database connection is there, can we just run `$wpdb->get_var( 'SELECT VERSION()' )` and watch for a non-`null` value?

Alternatively, we could query for a non-existent value like `$wpdb->get_var( $wpdb->prepare( "SELECT option_id FROM $wpdb->options WHERE option_name = %s", 'vip_healthcheck_' . rand() ) )` and verify that `$wpdb->last_error` is empty.

This way, we can avoid having to add the option.